### PR TITLE
Sd/v05

### PR DESCRIPTION
**Pull request by SimonDanisch**
* add build script (tested on win10)
* some more 0.5 compats

Jeez, we really need to switch to defining unsafe_convert methods etc...

### CODE REVIEW

**Comment by SimonDanisch** [comment]
@maleadt @vchuravy can we have Tim CI? :)

**Comment by maleadt** [comment]
@SimonDanisch not sure why it didn't update... but I've activated CI, build on master succeeds https://ci.maleadt.net/buildbot/julia/builders/CLBLAS.jl%3A%20Julia%200.5%20%28x86-64%29/builds/1 but fails on this PR https://ci.maleadt.net/buildbot/julia/builders/CLBLAS.jl%3A%20Julia%200.5%20%28x86-64%29/builds/2

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    if Sys.ARCH == :x86_64
+        uri = URI(baseurl * "Linux-x64.tar.gz")
+        provides(
+            Binaries, uri,
+            libCLBLAS, unpacked_dir = archive,
+            installed_libpath = libpath, os = :Linux
+        )
+    end
+end
+if is_apple()
```
Comment (SimonDanisch) [inline deps/build.jl]: Yeah, it shouldn't ;)

File: examples/AddNumbers.jl [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import OpenCL
-const cl = OpenCL
+using OpenCL: cl
```
Comment (vchuravy) [inline examples/AddNumbers.jl]: `using OpenCL` should be enough

**Comment by SimonDanisch** [comment]
I use the prebuild binaries installed with apt-get

**Comment by maleadt** [comment]
clblas binaries on my CI are the ones from GitHub too (Debian stable doesn't offer them).

**Comment by SimonDanisch** [comment]
I'd like to leave the failures for another PR, since it should still be a strict improvement and it seems as if tests haven't been passing previously....

**Comment by dfdx** [comment]
No objections from my side. I believe we should currently pay more attention to GPUArrays anyway, the future of CLBLAS heavily depends on it. I expect we will still link to libclblas in some way, but in the end bindings and installation procedure might be quite different.

**Comment by SimonDanisch** [comment]
Great!
Although I'm wondering a bit what's going on here... Were the tests actually passing at any point (locally) ?

**Comment by dfdx** [comment]
I checked CLBLAS's master a week ago on my Ubuntu box and it worked perfectly well. I can check this PR's branch in the evening.

**Comment by dfdx** [comment]
Tests pass fine on this branch with CUDA and GeForce GTX 960M **given** that `Complex64` is excluded from tested types. Since this is a known and documented issue, I believe we can exclude this type and merge the PR.

**Comment by SimonDanisch** [comment]
Okay, I think this should be ready to go!

**Comment by vchuravy** [comment]
Tests for `gemv`? Otherwise LGTM

**Comment by SimonDanisch** [comment]
yeah they bring me a bit in a code duplication issue, since I now started to test them in GPUArrays...
I guess the way forward is to rip all the high level stuff out of CLBLAS and do testing etc in GPUArrays

**Comment by vchuravy** [comment]
We still need to have some minimal tests here. But that shouldn't hold up
this PR.

**Comment by SimonDanisch** [comment]
sweet! :)

**Pull request merged**